### PR TITLE
Make ldmsd_stream_publish_file() support Piped stdin

### PR DESCRIPTION
**Pull request by nichamon**

### CODE REVIEW

**Comment by nichamon** [comment]
I have tested the patch with.
```
echo -n pause | ldmsd_stream_publish ....
```
and
```
ldmsd_stream_publish ... -f <file>
```

LDMSD receives the stream successfully.

**Comment by nichamon** [comment]
I also think about limiting the size of a stream if `pipe` is used. The limit could be the max message length, or it could be a constant number, not varied by transports. Limiting the size would prevent the out-of-memory error. What do you guys think?

File: ldms/src/ldmsd/ldmsd_stream.c [hunk 1]
```diff
@@ -427,23 +427,60 @@ int ldmsd_stream_publish_file(const char *stream, const char *type,
...
+
+	} else {
+		if (ESPIPE != errno) {
+			msglog("The given file is invalid.\n");
+			goto close_xprt;
+		}
+
+		/* a non-seekable FILE */
+		char *s;
+		char b[max_msg_len];
```
Comment (tom95858) [inline ldms/src/ldmsd/ldmsd_stream.c:460]: This buffer needs to be malloc'd to avoid blowing out the stack.

**Review by tom95858 — CHANGES_REQUESTED** [review]

File: ldms/src/ldmsd/ldmsd_stream.c (hunk 1 above)
Comment (nichamon) [reply ldms/src/ldmsd/ldmsd_stream.c]: Got it!

**Pull request merged**